frontend-service/src/main/java/com/amazonaws/blox/frontend/mappers/CreateEnvironmentMapper.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.frontend.mappers;
+
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.CreateEnvironmentRequest;
+import com.amazonaws.blox.frontend.operations.CreateEnvironment;
+import com.amazonaws.serverless.proxy.internal.model.ApiGatewayRequestContext;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(uses = InstanceGroupMapper.class)
+public interface CreateEnvironmentMapper {
+  @Mapping(target = "environmentId.accountId", source = "context.accountId")
+  @Mapping(target = "environmentId.cluster", source = "cluster")
+  @Mapping(target = "environmentId.environmentName", source = "request.environmentName")
+  CreateEnvironmentRequest toDataServiceRequest(
+      ApiGatewayRequestContext context,
+      String cluster,
```

Comment:
createenvrequest doesn't contain cluster?

Reply:
Nope, it's a Spring MVC `@PathVariable`, since it's part of the URL:

https://github.com/wjbuys/blox/blob/dcc96ab5a9d5d4d36001db984acab2aab47c4075/frontend-service/src/main/java/com/amazonaws/blox/frontend/operations/CreateEnvironment.java#L45-L55

I originally tried to move the `@PathVariable String cluster` argument into the `CreateEnvironmentRequest` object as a field. However, even though the Spring MVC docs suggest that this will correctly be bound, the Swagger generator we're using doesn't support it, and totally omits `cluster` from the generated Swagger model, and so we don't get it in the generated client.